pages/items/[itemId].tsx

```diff
@@ -0,0 +1,132 @@
+import { ChangeEvent, useEffect, useRef, useState } from "react";
+import { GetServerSidePropsContext } from "next";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import Btn from "@/components/Btn";
+import BtnImage from "@/components/BtnImage";
+import CheckListDetail from "@/components/CheckListDetail";
+import Gnb from "@/components/Gnb";
+import {
+  deleteItem,
+  getItem,
+  patchItem,
+  postImage,
+  ResponseItem,
+} from "@/lib/api";
+import ic_img from "@/assets/images/img.png";
+import ic_memo from "@/assets/images/memo.png";
+import styles from "@/styles/item.module.css";
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const itemId = Number(context.params?.itemId);
+  const item = await getItem(itemId);
+
+  return { props: { item, itemId } };
+}
+
+export default function Item({
+  item,
+  itemId,
+}: {
+  item: ResponseItem;
+  itemId: number;
+}) {
+  const [imageUrl, setImageUrl] = useState(item.imageUrl);
+  const [disabled, setDisabled] = useState(false);
+  const formRef = useRef<HTMLFormElement>(null);
+  const router = useRouter();
+
+  const handleInputChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      try {
+        const { url } = await postImage(e.target.files[0]);
+        setImageUrl(url);
+      } catch (error) {
+        alert((error as Error).message);
+      }
+    }
+  };
+
+  const handleTextareaChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    e.target.style.height = "auto";
+    e.target.style.height = Math.min(e.target.scrollHeight, 229) + "px";
+  };
+
+  const handleEditClick = async () => {
+    try {
+      setDisabled(true);
+      const data = Object.fromEntries(
+        new FormData(formRef.current ?? undefined).entries()
+      );
+      await patchItem(itemId, { ...data, imageUrl: imageUrl ?? "" });
+      router.push("/");
+    } catch (error) {
+      alert((error as Error).message);
+      setDisabled(false);
```

Comment:
💊 제안
타입 단언문대신 아래처럼 타입가드를 통해 작성하시는 것을 추천드립니다.

```suggestion
        if (error instanceof Error) {
          alert(error.message);
        }
      setDisabled(false);
```